workflows/sarek/main.nf

```diff
@@ -149,8 +156,59 @@ workflow SAREK {
+            // TO-DO: Would it perhaps be better to also call flowcellLaneFromFastq(files[1]) and check that we get the same flowcell-id?
+
+            // Don't use a random element for ID, it breaks resuming
+            def read_group = "\"@RG\\tID:${flowcell}.${meta.sample}.${meta.lane}\\t${CN}PU:${meta.lane}\\tSM:${meta.patient}_${meta.sample}\\tLB:${meta.sample}\\tDS:${params.fasta}\\tPL:${params.seq_platform}\""
+            meta  = meta - meta.subMap('lane') + [read_group: read_group.toString()]
+            return [ meta, files ]
+        }
+
+        // Just one fastq.gz.spring-file with both R1 and R2
+        fastq_gz_pair_from_spring = SPRING_DECOMPRESS_TO_FQ_PAIR(input_sample_type.one_fastq_gz_spring)
+
+        one_fastq_gz_from_spring = fastq_gz_pair_from_spring.fastq.map { meta, files ->
+            def CN = params.seq_center ? "CN:${params.seq_center}\\t" : ''
+
+            // Here we're assuming that fastq_1 and fastq_2 are from the same flowcell:
+            def flowcell = flowcellLaneFromFastq(files[0])
+            // TO-DO: Would it perhaps be better to also call flowcellLaneFromFastq(files[1]) and check that we get the same flowcell-id?
+
+            // Don't use a random element for ID, it breaks resuming
+            def read_group = "\"@RG\\tID:${flowcell}.${meta.sample}.${meta.lane}\\t${CN}PU:${meta.lane}\\tSM:${meta.patient}_${meta.sample}\\tLB:${meta.sample}\\tDS:${params.fasta}\\tPL:${params.seq_platform}\""
+            meta  = meta - meta.subMap('lane') + [read_group: read_group.toString()]
+            return [ meta, files ]
+        }
+
+        // Two fastq.gz.spring-files - one for R1 and one for R2
+        R1_fastq_gz_from_spring = SPRING_DECOMPRESS_TO_R1_FQ_PAIR(input_sample_type.two_fastq_gz_spring.map{ meta, files -> [meta, files[0] ]})
+        R2_fastq_gz_from_spring = SPRING_DECOMPRESS_TO_R2_FQ_PAIR(input_sample_type.two_fastq_gz_spring.map{ meta, files -> [meta, files[1] ]})
+
+        two_fastq_gz_from_spring = R1_fastq_gz_from_spring.fastq.join(R2_fastq_gz_from_spring.fastq).map{ meta, fastq_1, fastq_2 -> [meta, [fastq_1, fastq_2]]}
+
+        two_fastq_gz_from_spring = two_fastq_gz_from_spring.map { meta, files ->
+            def CN = params.seq_center ? "CN:${params.seq_center}\\t" : ''
+
+            // Here we're assuming that fastq_1 and fastq_2 are from the same flowcell:
+            def flowcell = flowcellLaneFromFastq(files[0])
+            // TO-DO: Would it perhaps be better to also call flowcellLaneFromFastq(files[1]) and check that we get the same flowcell-id?
+
+            // Don't use a random element for ID, it breaks resuming
+            def read_group = "\"@RG\\tID:${flowcell}.${meta.sample}.${meta.lane}\\t${CN}PU:${meta.lane}\\tSM:${meta.patient}_${meta.sample}\\tLB:${meta.sample}\\tDS:${params.fasta}\\tPL:${params.seq_platform}\""
+            meta  = meta - meta.subMap('lane') + [read_group: read_group.toString()]
+            return [ meta, files ]
         }
```

Comment:
Is there a way to avoid so much code duplication? can we function it out?

Reply:
Good point about the code duplication. It would be nice to get rid of that. Should I try do that "map" as a function at the bottom of the file?  Around [here](https://github.com/asp8200/sarek/blob/1a649a7fd4ead0c5d2555682c514daaa5f71a48e/workflows/sarek/main.nf#L966)?

Reply:
Ok, I introduced the function `addReadgroupToMeta` to get rid of the code-duplication.